feat(linter): add support for `test.for` in vitest

camchenry opened the pull request:
- Closes https://github.com/oxc-project/oxc/issues/12152

Now that we don't have a huge list of valid function calls, adding support for this alias is actually pretty easy.

camc314 commented:
### Merge activity

* **Dec 16, 9:09 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 16, 9:10 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

camchenry committed:
feat(linter): add support for `test.for` in vitest (#16925)

- Closes https://github.com/oxc-project/oxc/issues/12152

Now that we don't have a huge list of valid function calls, adding support for this alias is actually pretty easy.
